common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -0,0 +1,71 @@
+"""
+Viewset for auth/saml/v0/samlproviderconfig
+"""
+
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from ..models import SAMLProviderConfig
+from .serializers import SAMLProviderConfigSerializer
+
+
+class SAMLProviderMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderConfigSerializer
+
+
+class SAMLProviderConfigViewSet(PermissionRequiredMixin, SAMLProviderMixin, viewsets.ModelViewSet):
+    """
+    A View to handle SAMLProviderConfig CRUD
+
+    Usage:
+        [HttpVerb] /auth/saml/v0/provider_config/?enterprise-id=uuid
```

Comment:
I'd also expect us to have this access pattern:

```
GET /auth/saml/v0/provider_config/:id
PUT /auth/saml/v0/provider_config/:id {putData}
PATCH /auth/saml/v0/provider_config/:id {patchData}
DELETE /auth/saml/v0/provider_config/:id
```

If we're operating on an admin management style page, when querying for the data on a dashboard view we'll want to query for a these via enterprise_id which will power our list view.

But we also need to consider the cases where we are operating on these configs. The specific config view form page for example I'd imagine would be surfaced with a specific GET call to a primary key, and then either PUT/PATCH/DELETE against the same primary key. We already have that data when we access these forms, there shouldn't be any need to do an extra jump and derive what object we're operating against.

Reply:
indeed, thatis the case, I should update this doc to reflect that, thanks

Reply:
at the risk of a slightly confusing/incomplete api, I think the GET by :pk will not yet work because I am always requiring the enterprise_customer_uuid.. I think it can be fixed, but want to punt on that for the time being if that is acceptable ... looks like the frontend will always supply the enterprise_id as of this time ... @taliaedX 

I updated the doc, for rest of the cases that apply